### PR TITLE
Chore/promise rejection

### PR DESCRIPTION
**Pull request by gairTanm**
**What kind of change does this PR introduce?**
Handles promise rejection

**Did you add tests for your changes?**
None required (I suppose)
**If relevant, did you update the documentation?**
Not required
**Summary**
Added a .catch block.

**Does this PR introduce a breaking change?**
No

**Other information**
First PR, tried following the contributing guide :smile:

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -50,8 +48,3 @@ packages/**/*.map
 # cache
 .eslintcache
```
Comment (evilebottnawi) [inline .gitignore:49]: Do not touch this file, if you need cleanup or found unnecessary things, let's do it in the separate PR

File: packages/serve/src/index.ts [hunk 2]
```diff
@@ -15,7 +15,11 @@ export default function serve(...args: string[]): void {
...
 
-    cli.getCompiler(webpackArgs, core).then((compiler): void => {
-        startDevServer(compiler, devServerArgs);
-    });
+    cli.getCompiler(webpackArgs, core)
+        .then((compiler): void => {
+            startDevServer(compiler, devServerArgs);
+        })
+        .catch((err: object): void => {
+            console.error(err);
```
Comment (evilebottnawi) [inline packages/serve/src/index.ts:23]: We should `throw` error, not logging

**Review by anshumanv — COMMENTED** [review]
You have unrelated commits, please remove them

**Review by anshumanv — CHANGES_REQUESTED** [review]
your base branch is wrong

**Comment by gairTanm** [comment]
@anshumanv I have just that one commit, the catch block. Also, I committed those changes on a separate branch. Do let me know if I'm missing something.

File: packages/serve/src/index.ts (hunk 2 above)
Comment (gairTanm) [reply packages/serve/src/index.ts:23]: So should I just throw an error, or log the err object and then throw the error?

**Comment by anshumanv** [comment]
@gairTanm you have one commit because @snitin315 fixed your branch, you should PR to `next` branch

**Comment by snitin315** [comment]
@gairTanm , I have updated the base branch to `next`.

File: packages/serve/src/index.ts (hunk 2 above)
Comment (anshumanv) [reply packages/serve/src/index.ts:23]: Just throw with error message

**Review by anshumanv — COMMENTED** [review]
Can you also check if there are any other uncaught promises?

**Comment by gairTanm** [comment]
> Can you also check if there are any other uncaught promises?

will do

**Comment by gairTanm** [comment]
Can I start a new pr? I had to reset my fork :disappointed: . I'll check for other uncaught promises too

**Comment by anshumanv** [comment]
Reset your fork meaning? I still see your commits here

**Comment by gairTanm** [comment]
I deleted my fork, ran into some problems (of my own making) :disappointed:. Should I move with that new pr?

**Comment by anshumanv** [comment]
Sure. Closing this one.
